pilot/pkg/model/push_context.go

```diff
@@ -766,24 +766,19 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
-// Services returns the list of services that are visible to a Proxy in a given config namespace
-func (ps *PushContext) Services(proxy *Proxy) []*Service {
-	// If proxy has a sidecar scope that is user supplied, then get the services from the sidecar scope
-	// sidecarScope.config is nil if there is no sidecar scope for the namespace
-	if proxy != nil && proxy.SidecarScope != nil && proxy.Type == SidecarProxy {
-		return proxy.SidecarScope.services
-	}
-
+// servicesExportToNamespace returns the list of services that are visible to a namespace.
+// namespace "" indicates all namespaces
+func (ps *PushContext) servicesExportToNamespace(ns string) []*Service {
```

Comment:
nit: servicesExportToNamespace -> servicesExportedToNamespace

Reply:
why do not we just use the existing Services method and change it arg to take namespace?

Reply:
It should at least be private